lib/cog/repository/groups.ex

```diff
@@ -0,0 +1,215 @@
+defmodule Cog.Repository.Groups do
+  @moduledoc """
+  Behavioral API for interacting with user groups. Prefer these
+  functions over direct manipulation with 'Cog.Repo'.
+  """
+
+  alias Cog.Repo
+  alias Cog.Models.Group
+  alias Cog.Models.User
+  alias Cog.Models.Role
+  import Ecto.Query, only: [from: 1, from: 2]
+
+
+  @doc """
+  Creates a new user group given a map of attributes
+  """
+  @spec new(Map.t) :: {:ok, %Group{}} | {:error, Ecto.Changeset.t}
+  def new(attrs) do
+    %Group{}
+    |> Group.changeset(attrs)
+    |> Repo.insert
+  end
+
+  @doc """
+  Retrieves all user groups
+  """
+  @spec all :: [%Group{}]
+  def all,
+    do: Repo.all(Group) |> Repo.preload([:direct_user_members, :direct_group_members, :roles])
```

Comment:
Ah, nevermind... I see you're just pulling the logic from the controllers into a repository. There's still a lot of that authz infrastructure we need to yank out, so we can just leave this until then.